Compfiles/Imo1970P4.lean

```diff
@@ -21,15 +20,937 @@ equals the product of the nubmers in the other set.
 
 namespace Imo1970P4
 
-determine SolutionSet : Finset ℕ+ := sorry
+snip begin
+
+lemma card_opposite (s s' s'' : Finset ℕ) (predicate: ℕ → Prop) [DecidablePred predicate] (filter : s' = (s.filter (λ x => predicate x)))
+                    (opposite_filter: s'' = (s.filter (λ x => ¬ predicate x))) : s'.card + s''.card = s.card := by
+  rw[filter]
+  rw[opposite_filter]
+  have := @Finset.filter_card_add_filter_neg_card_eq_card ℕ s predicate
+  apply this
+
+lemma dvd_prod_of_dvd_elem (p : ℕ) (s : Finset ℕ) (x : ℕ) (H : p ∣ x) : (x ∈ s) → p ∣ (∏ m ∈ s, m) := by
```

Comment:
See [`Finset.dvd_prod_of_mem`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/BigOperators/Group/Finset.html#Finset.dvd_prod_of_mem).

Reply:
Oh, I see you're using that below. But I think you shouldn't even need this lemma, and ought to be able to use `Finset.dvd_prod_of_mem` directly.

Reply:
OK I've used it in three other places, but managed to remove it, i'll push the changes in a minute.